net/shadowsocks/encryption.cc

```diff
@@ -258,5 +258,57 @@ void EncryptedStream::write_payload(
         });
 }
 
+EncryptedDatagram::EncryptedDatagram(
+    udp::socket &socket, const MasterKey &master_key)
+    : socket_(socket),
+      master_key_(master_key),
+      read_buffer_(std::make_unique<uint8_t[]>(read_buffer_size_)),
+      write_buffer_(std::make_unique<uint8_t[]>(write_buffer_size_)) {}
+
+void EncryptedDatagram::receive_from(
+    std::function<void(
+        std::error_code, absl::Span<const uint8_t>, 
+        const udp::endpoint&)> callback) {
+    socket_.async_receive_from(
+        boost::asio::buffer(read_buffer_.get(), read_buffer_size_), ep_,
```

Comment:
应使用`net::buffer` instead of `boost::asio::`，并且由于这个library在`net` namespace里，因此获得了不需要`net::`前缀的_红利_

https://github.com/iceboy233/trunk/blob/master/net/asio.h